frontend/packages/dev-console/src/components/pipelines/modals/triggers/resource-utils.ts

```diff
@@ -0,0 +1,79 @@
+import { getRandomChars } from '@console/shared';
+import { apiVersionForModel, RouteKind } from '@console/internal/module/k8s';
+import { RouteModel } from '@console/internal/models';
+import { EventListenerModel, TriggerTemplateModel } from '../../../../models';
+import { PipelineRun } from '../../../../utils/pipeline-augment';
+import {
+  TriggerBindingKind,
+  EventListenerKind,
+  TriggerTemplateKind,
+  TriggerTemplateKindParam,
+} from '../../resource-types';
+
+export const createTriggerTemplate = (
+  pipelineRun: PipelineRun,
+  params: TriggerTemplateKindParam[],
+): TriggerTemplateKind => {
+  return {
+    apiVersion: apiVersionForModel(TriggerTemplateModel),
+    kind: TriggerTemplateModel.kind,
+    metadata: {
+      name: `trigger-template-${pipelineRun.metadata.name}`,
+    },
+    spec: {
+      params,
+      // TODO: Add PipelineResources instead of referencing them from PipelineRun
+      resourcetemplates: [pipelineRun],
+    },
+  };
+};
+
+export const createEventListener = (
+  triggerBindings: TriggerBindingKind[],
+  triggerTemplate: TriggerTemplateKind,
+): EventListenerKind => {
+  return {
+    apiVersion: apiVersionForModel(EventListenerModel),
+    kind: EventListenerModel.kind,
+    metadata: {
+      name: `event-listener-${getRandomChars()}`,
+    },
+    spec: {
+      triggers: [
+        {
+          bindings: triggerBindings.map(({ metadata: { name } }) => ({ name })),
+          template: { name: triggerTemplate.metadata.name },
+        },
+      ],
+    },
+  };
+};
+
+export const createEventListenerRoute = (
+  eventListener: EventListenerKind,
+  targetPort: number = 8080,
+): RouteKind => {
+  const eventListenerName = eventListener.metadata.name;
+  return {
+    apiVersion: apiVersionForModel(RouteModel),
+    kind: RouteModel.kind,
+    metadata: {
+      name: eventListenerName,
+      labels: {
+        'app.kubernetes.io/managed-by': EventListenerModel.kind,
+        'app.kubernetes.io/part-of': 'Triggers',
```

Comment:
Should the application name will always be `Triggers`?

Reply:
Oh good call... I can probably improve this. This was a copy of my example YAML that I just slapped variables into -- I'll look at this more critically.